src/App.tsx

```diff
@@ -121,23 +121,46 @@ function App(): ReactElement {
   // UTILITY METHODS /////////////////////////////////////////////////////////////
 
   /**
-   * Get a set of URL parameters that represent the current collection, dataset, feature, track,
-   * and frame information. (Convenience wrapper for `urlUtils.getUrlParams`.)
+   * Get a URL query string representing the current collection, dataset, feature, track,
+   * and frame information.
    */
   const getUrlParams = useCallback((): string => {
     let datasetParam: string | undefined = datasetKey;
-    if (!collection?.url) {
-      // The collection has no source file; use the dataset URL instead
+    let collectionParam = collection?.url;
+    // If there is no collection loaded (e.g. a single dataset URL was loaded),
+    // use the dataset URL instead.
+    if (collectionParam === null) {
       datasetParam = dataset?.manifestUrl;
+      collectionParam = undefined;
     }
```

Comment:
could this be moved into `urlUtils.stateToUrlQueryString` ?

Reply:
or if you want to keep that as something that cleanly only takes in a `state` parameter, a new function on `urlUtils` called `convertParamsToState` that returns the state object

Reply:
Hmm, I don't think there's a clean way to move this into `urlUtils` without introducing a lot of fragile boilerplate, since so much of the decisionmaking here is dependent on the way state is stored/represented in App...

ex: if I added a `convertParamsToState(rawParams: ParamType)` to `url_utils` I'd have to add the `ParamType` with an entry for every state variable in App.

Reply:
Made some changes, let me know if that looks better?